libraries/chain/genesis/initializators/rewards_initializator.cpp

```diff
@@ -73,14 +75,19 @@ void rewards_initializator_impl::create_balancers(initializator_context& ctx)
 
 void rewards_initializator_impl::create_fund_budget(initializator_context& ctx)
 {
-    dynamic_global_property_service_i& dgp_service = ctx.services().dynamic_global_property_service();
-    budget_service_i& budget_service = ctx.services().budget_service();
+    dynamic_global_property_service_i& dprops_service = ctx.services().dynamic_global_property_service();
+    fund_budget_service_i& budget_service = ctx.services().fund_budget_service();
+
+    FC_ASSERT(!budget_service.is_exists());
 
-    FC_ASSERT(!budget_service.is_fund_budget_exists());
+    auto balance = asset(ctx.genesis_state().rewards_supply.amount, SP_SYMBOL);
 
-    fc::time_point deadline = dgp_service.get_genesis_time() + fc::days(SCORUM_REWARDS_INITIAL_SUPPLY_PERIOD_IN_DAYS);
+    time_point_sec start_date = dprops_service.get_genesis_time();
+    fc::time_point deadline
+        = dprops_service.get_genesis_time() + fc::days(SCORUM_REWARDS_INITIAL_SUPPLY_PERIOD_IN_DAYS);
```

Comment:
Use `start_date` from previous line instead of `dprops_service.get_genesis_time()`